Mathlib/Order/CompactlyGenerated/Basic.lean

```diff
@@ -585,13 +585,13 @@ Most explicitly, every element is the complement of a supremum of indepedendent
 
 /-- In an atomic lattice, every element `b` has a complement of the form `sSup s`, where each
 element of `s` is an atom. See also `complementedLattice_of_sSup_atoms_eq_top`. -/
-theorem exists_sSupIndep_isCompl_sSup_atoms (h : sSup { a : α | IsAtom a } = ⊤) (b : α) :
-    ∃ s : Set α, sSupIndep s ∧
-    IsCompl b (sSup s) ∧ ∀ ⦃a⦄, a ∈ s → IsAtom a := by
+theorem exists_sSupIndep_isCompl_sSup_atoms (b c : α) (hbc : b ≤ c)
+    (h : sSup {a ≤ c | IsAtom a} = c) :
+    ∃ s : Set α, sSupIndep s ∧ Disjoint b (sSup s) ∧ b ⊔ sSup s = c ∧ ∀ ⦃a⦄, a ∈ s → IsAtom a := by
```

Comment:
The docstring and name for this lemma no longer seem accurate, since the new result isn't quite a complement. Also, what do you think of keeping the old version around (as a one-line corollary of your new one), since it's a useful special case? (This also avoids the deprecation!)